submission/loglist_refresher_test.go

```diff
@@ -124,7 +126,13 @@ func TestNewLogListRefresher(t *testing.T) {
 			} else if gotErr && !tc.errRegexp.MatchString(err.Error()) {
 				t.Fatalf("llr.Refresh() = (_, %q), want err to match regexp %q", err, tc.errRegexp)
 			}
-			if diff := cmp.Diff(ll, tc.wantLl); diff != "" {
+			if (ll == nil) != (tc.wantLl == nil) {
+				t.Fatalf("llr.Refresh() = (%v, _), expected value? %t", ll, tc.wantLl != nil)
+			}
+			if ll == nil {
+				return
+			}
+			if diff := cmp.Diff(ll.List, tc.wantLl); diff != "" {
```

Comment:
Also test that `ll.Source` and `ll.When` are set correctly?

Reply:
Added checks into tests.